orders.h

```diff
@@ -197,210 +202,220 @@ class SellOrder : public Order {
 
 class AttackOrder : public Order {
   public:
-    AttackOrder();
-    ~AttackOrder();
+    AttackOrder() : Order(O_ATTACK) {}
+    virtual ~AttackOrder();
 
     std::list<UnitId *> targets;
 };
 
 class BuildOrder : public Order {
   public:
-    BuildOrder();
-    ~BuildOrder();
+    BuildOrder(UnitId *target = nullptr) : Order(O_BUILD), target(target), needtocomplete(0) {}
+    BuildOrder(int needtocomplete) : Order(O_BUILD), target(nullptr), needtocomplete(needtocomplete) {}
+    virtual ~BuildOrder();
 
     UnitId *target;
-    int new_building;
+    int new_building = -1;
     int needtocomplete;
-    bool until_complete;
+    bool until_complete = false;
 };
 
 class SailOrder : public Order {
   public:
-    SailOrder();
-    ~SailOrder();
+    SailOrder() : Order(O_SAIL) {}
+    virtual ~SailOrder();
 
     std::list<MoveDir *> dirs;
 };
 
 class FindOrder : public Order {
   public:
-    FindOrder();
-    ~FindOrder();
+    FindOrder(int find) : Order(O_FIND), find(find) {}
+    virtual ~FindOrder() = default;
 
     int find;
 };
 
 class StealthOrder : public Order {
   public:
-    StealthOrder();
-    ~StealthOrder();
+    StealthOrder(int type = 0, UnitId *target = nullptr);
```

Comment:
Curious why you kept this one in the .cpp file and not moved it here much like you did the others?

Reply:
FYI, this is not a request to do it now, and I am fine leaving it as it is with this changes, just thinking out loud.

Reply:
Having a big "types.h" file is a code smell, yes. I would break that file up like you suggest, but maybe not today.